lib/registries/addon/helpers/math.ts

```diff
@@ -0,0 +1,96 @@
+import Helper from '@ember/component/helper';
+import { htmlSafe } from '@ember/string';
+import katex, { KatexRenderOptions } from 'katex';
+
+export interface Delimiter {
+    start: string;
+    end: string;
+    inline: boolean;
+}
+
+interface Match {
+    start: number;
+    end: number;
+    delimiter: Delimiter;
+}
+
+// Adapted from https://github.com/Khan/KaTeX/blob/master/contrib/auto-render/splitAtDelimiters.js
+export function findEnd(content: string, start: number, end: string) {
+    let contextLevel = 0;
+
+    for (let i = start; i < content.length; i++) {
+        if (contextLevel < 1 && content.startsWith(end, i)) {
+            return i + end.length;
+        } else if (content[i] === '}') {
+            contextLevel++;
+        } else if (content[i] === '{') {
+            contextLevel--;
```

Comment:
If I understand the intent, I think you got these flipped. When it finds a `{`, should be `contextLevel++` to prevent matching an end delimiter inside brackets.
```
just text $$ math! { $$ math } still math $$ okay text again
```